File: backend/app/schemas/pulse_inbox.py
from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel


class InboxAction(str, Enum):
    to_task = "to_task"
    to_note = "to_note"
    skip = "skip"


class InboxItemResponse(BaseModel):
    id: int
    text: Optional[str] = None
    sender_name: Optional[str] = None
    message_date: Optional[datetime] = None
    source_title: Optional[str] = None
    source_id: int
    ai_classification: Optional[str] = None
    ai_relevance: Optional[float] = None
    status: str
    collected_at: datetime

    model_config = {"from_attributes": True}


class InboxListResponse(BaseModel):
    items: list[InboxItemResponse]
    total: int


class InboxActionRequest(BaseModel):
    action: InboxAction


class BulkActionRequest(BaseModel):
    message_ids: list[int]
    action: InboxAction

File: frontend/src/types/pulse-inbox.ts
export interface InboxItem {
  id: number;
  text: string | null;
  sender_name: string | null;
  message_date: string | null;
  source_title: string | null;
  source_id: number;
  ai_classification: string | null;
  ai_relevance: number | null;
  status: string;
  collected_at: string;
}

export interface InboxListResponse {
  items: InboxItem[];
  total: number;
}

export type InboxAction = "to_task" | "to_note" | "skip";

export interface InboxActionRequest {
  action: InboxAction;
}

export interface BulkActionRequest {
  message_ids: number[];
  action: InboxAction;
}

export interface InboxActionResponse {
  status: string;
  action: string;
  message_id?: number;
  processed?: number;
}
